services/src/main/java/org/keycloak/broker/saml/SAMLIdentityProviderConfig.java

```diff
@@ -335,11 +362,169 @@ public void setSignSpMetadata(boolean signSpMetadata) {
+        getConfig().put(MDUI_PRIVACY_STATEMENT_URL, mduiPrivacyStatementURL);
+    }
+
+    public String getMduiLogo() {
+        return getConfig().get(MDUI_LOGO);
+    }
+
+    public void setMduiLogo(String mduiLogo) {
+        getConfig().put(MDUI_LOGO, mduiLogo);
+    }
+
+    public Integer getMduiLogoHeight() {
+        return getConfig().get(MDUI_LOGO_HEIGHT) == null ? null : Integer.valueOf(getConfig().get(MDUI_LOGO_HEIGHT));
+    }
+
+    public void setMduiLogoHeight(Integer mduiLogoHeight) {
+        getConfig().put(MDUI_LOGO_HEIGHT, String.valueOf(mduiLogoHeight));
+    }
+
+    public Integer getMduiLogoWidth() {
+        return getConfig().get(MDUI_LOGO_WIDTH) == null ? null : Integer.valueOf(getConfig().get(MDUI_LOGO_WIDTH));
+    }
+
+    public void setMduiLogoWidth(Integer mduiLogoWidth) {
+        getConfig().put(MDUI_LOGO_WIDTH, String.valueOf(mduiLogoWidth));
+    }
+
+    public String getMdOrganizationName() {
+        return getConfig().get(MD_ORGANIZATION_NAME);
+    }
+
+    public void setMdOrganizationName(String mdOrganizationName) {
+        getConfig().put(MD_ORGANIZATION_NAME, mdOrganizationName);
+    }
+
+    public String getMdOrganizationDisplayName() {
+        return getConfig().get(MD_ORGANIZATION_DISPLAY_NAME);
+    }
+
+    public void setMdOrganizationDisplayName(String mdOrganizationDisplayName) {
+        getConfig().put(MD_ORGANIZATION_DISPLAY_NAME, mdOrganizationDisplayName);
+    }
+
+    public String getMdOrganizationURL() {
+        return getConfig().get(MD_ORGANIZATION_URL);
+    }
+
+    public void setMdOrganizationURL(String mdOrganizationURL) {
+        getConfig().put(MD_ORGANIZATION_URL, mdOrganizationURL);
+    }
+
+    public String getMdContactType() {
+        return getConfig().get(MD_CONTACT_TYPE);
+    }
+
+    public void setMdContactType(String mdContactType) {
+        getConfig().put(MD_CONTACT_TYPE, mdContactType);
+    }
+
+    public String getMdContactCompany() {
+        return getConfig().get(MD_CONTACT_COMPANY);
+    }
+
+    public void setMdContactCompany(String mdContactCompany) {
+        getConfig().put(MD_CONTACT_COMPANY, mdContactCompany);
+    }
+
+    public String getMdContactGivenName() {
+        return getConfig().get(MD_CONTACT_GIVEN_NAME);
+    }
+
+    public void setMdContactGivenName(String mdContactGivenName) {
+        getConfig().put(MD_CONTACT_GIVEN_NAME, mdContactGivenName);
+    }
+
+    public String getMdContactSurname() {
+        return getConfig().get(MD_CONTACT_SURNAME);
+    }
+
+    public void setMdContactSurname(String mdContactSurname) {
+        getConfig().put(MD_CONTACT_SURNAME, mdContactSurname);
+    }
+
+    public String getMdContactEmailAddress() {
+        return getConfig().get(MD_CONTACT_EMAIL_ADDRESS);
+    }
+
+    public void setMdContactEmailAddress(String mdContactEmailAddress) {
+        getConfig().put(MD_CONTACT_EMAIL_ADDRESS, mdContactEmailAddress);
```

Comment:
How will this be implemented in ui taking into account that config in IdentityProviderRepresentation is Map<String,String>? User should not edit json.
That's why I concat list to String with comma seperator.

Reply:
Joining the attributes or even assuming that they are serializable to `String` is a wrong assumption. Let me reiterate:

> The way it is now completely ignores information about the attribute dataformat, and relies on uncertain fact that the attributes are of string type.

You need to introduce a SPI that would handle custom attributes. That way you can have the geant and other plugged in.

I agree that there is no expectation that user would edit JSON. How that is propagated to UI depends on how the SPI is designed, but can and has to be done purely in the SPI. Note that SPI can provide its partials which are then dynamically included in the UI (see `ng-include`), it can interact with UI via REST endpoints etc.

Reply:
Here, I am not refereing to attributes and proposed SPI solution. md:EmailAddress and md:TelephoneNumber are List<String> as you can see in [documentation](https://docs.oasis-open.org/security/saml/v2.0/saml-metadata-2.0-os.pdf) and in Keycloak code. In my last commit, I propose a solution based on your comment.

Reply:
Comma is a not often used but still valid character in email address. See https://tools.ietf.org/html/rfc5321#section-4.1.2. Do use JSON (and only plain, not pretty-printed which eats space in database without any value). Addresses like `"smith,jr."@domain.com` are valid.

Reply:
I have changed  JsonSerialization method